packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1305,6 +1305,25 @@ abstract class ResidentRunner {
     }
   }
 
+  Future<void> callConnectedVmServiceUriExtension() async {
+    await Future.wait(<Future<vm_service.Isolate>>[
+      for (final FlutterDevice device in flutterDevices)
+        waitForExtension(device.vmService, 'ext.flutter.connectedVmServiceUri'),
+    ]);
+    for (final FlutterDevice device in flutterDevices) {
+      try {
+        unawaited(invokeFlutterExtensionRpcRawOnFirstIsolate(
+          'ext.flutter.connectedVmServiceUri',
+          params: <String, dynamic>{
+            'value': device.vmService.httpAddress.toString(),
+          },
+        ));
+      } on Exception catch (e) {
+        globals.printError(e.toString());
```

Comment:
Some more context here could help: "Failed to set vm service URI, __ deeplinks will not show in flutter errors" or something like that.